fleet/fleet.go

```diff
@@ -0,0 +1,87 @@
+// Package fleet implements a fleet client providing basic operations against a
+// fleet endpoint through fleet's HTTP API. Higher level scheduling and
+// management should be built on top of that.
+package fleet
+
+import (
+	"net"
+	"net/http"
+
+	"github.com/coreos/fleet/client"
+	"github.com/coreos/fleet/etcd"
+	"github.com/coreos/fleet/machine"
+	"github.com/coreos/fleet/registry"
+	"github.com/coreos/fleet/schema"
+	"github.com/coreos/fleet/unit"
+)
+
+// TODO
+type Config struct{}
+
+// TODO
+func DefaultConfig() Config {
+	newConfig := Config{}
+
+	return newConfig
+}
+
+type Status struct {
+	// SystemdStatus represents the status within systemd.
+	SystemdStatus containerstates.State
+
+	// FleetStatus represents the status within fleet.
+	FleetStatus containerstates.State
+
+	// HostIP represents the hosts IP where the related unit is running on. If
+	// the unit is not scheduled, HostIP is empty.
+	HostIP net.IP
+}
+
+// Fleet defines the interface a fleet client needs to implement to provide
+// basic operations against a fleet endpoint.
+type Fleet interface {
```

Comment:
The package is called fleet. So it is about having an instance of the implementation the package provides. That is why I named it simply fleet. I prefer the short name `Fleet` though.